File: programmers/01.hash/완주하지 못한 선수/js/01.js
function solution(participant, completion) {
    const participantPerson = participant.sort();
    const completionPerson = completion.sort();

    return participantPerson.find((person, idx) => person !== completionPerson[idx]);
}

// test
solution(["leo", "kiki", "eden"], ["eden", "kiki"]); // "leo"
solution(
    ["marina", "josipa", "nikola", "vinko", "filipa"],
    ["josipa", "filipa", "marina", "nikola"]
); // "vinko"
solution(["mislav", "stanko", "mislav", "ana"], ["stanko", "ana", "mislav"]); // "mislav"

File: programmers/01.hash/완주하지 못한 선수/python/01.py
def solution(participant, completion):

    participant.sort()
    completion.sort()

    for i, person in enumerate(completion):
        if person != participant[i]:
            return participant[i]

    return participant[-1]
    
    
    
    



solution(['leo', 'kiki', 'eden'], ['eden', 'kiki'])
solution(['marina', 'josipa', 'nikola', 'vinko', 'filipa'], ['josipa', 'filipa', 'marina', 'nikola'])
solution(['mislav', 'stanko', 'mislav', 'ana'], ['stanko', 'ana', 'mislav'])